### ISSUE TITLE
Vertical scrollbar in child window doesn't affect content region width

### ISSUE THREAD

**Issue by Teselka** [ocornut/imgui#8195]
### Version/Branch of Dear ImGui:

master

### Back-ends:

imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp

### Compiler, OS:

Linux + GCC

### Full config/build information:

```
Dear ImGui 1.91.6 WIP (19152)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=201703
define: __linux__
define: __GNUC__=14
--------------------------------
io.BackendPlatformName: imgui_impl_glfw
io.BackendRendererName: imgui_impl_opengl3
io.ConfigFlags: 0x00000003
 NavEnableKeyboard
 NavEnableGamepad
io.ConfigNavCaptureKeyboard
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000000E
 HasMouseCursors
 HasSetMousePos
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1280.00,720.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 1.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

### Details:

I've made a simple sidebar by using a child window, but when the scrollbar appears, it doesn't change the content region width as it should when using a normal window. 

### Screenshots/Video:

Expected behavior (as in normal window):
![image](https://github.com/user-attachments/assets/01377915-4c69-47fa-9ecf-e53b42a92e10)

Current behavior (child window):
![image](https://github.com/user-attachments/assets/1d48869b-fdf0-4b32-849b-a31d62d02b4f)

### Minimal, Complete and Verifiable Example code:

```cpp
ImGui::SetNextWindowSize({ 500.f, 350.f }, ImGuiCond_Appearing);
ImGui::Begin("Child window scrollbar bug");
ImGui::BeginChild("##Sidebar", { 250.f, 0.f });
for (int i = 0; i < 20; i++) {
    ImGui::Text("Item %i", i);
    if (i % 5 == 0)
        ImGui::Separator();
}
ImGui::EndChild();
ImGui::End();
```

**Comment by ocornut** [comment]
> it doesn't change the content region width as it should when using a normal window.

I am not sure what you mean exactly?

If you mean that the issue is that the `Separator()` reach the right border touching the scrollbar, it is because a `BeginChild()` without borders by default has zero WindowPadding. If you enable borders `ImGuiChildFlags_Borders` by default it inherit WindowPadding:

![image](https://github.com/user-attachments/assets/cbf5a501-a932-43ae-9121-142d2c5f65c8)

If you use `ImGuiChildFlags_AlwaysUseWindowPadding`:

![image](https://github.com/user-attachments/assets/a7254e1e-661c-4f3f-abfe-89829aa939fd)

Then you have both the parent padding + the child padding.

**Comment by Teselka** [comment]
> If you mean that the issue is that the Separator() reach the right border touching the scrollbar, it is because a BeginChild() without borders by default has zero WindowPadding. If you enable borders ImGuiChildFlags_Borders by default it inherit WindowPadding:

Yes, thanks! I thought for some reason that scrollbar _overlaps_ it.
But now i have another issue: i don't really want window padding from every single side, except for the right (because it basically doubles padding amount)

**Comment by ocornut** [comment]
If you child window completely fill the parent window (and that's where the issue is most likely to happen) then there's no value in using a child window. What's your actual setup?

**Comment by Teselka** [comment]
> If you child window completely fill the parent window (and that's where the issue is most likely to happen) then there's no value in using a child window. What's your actual setup?

It'a sidebar menu

![image](https://github.com/user-attachments/assets/d5f59a87-69bc-4809-b2e4-25992d4852cf)

**Comment by ocornut** [comment]
I realize it is not ideal, but the easiest way at the moment would be to "undo" the current offset in the parent window and then submit the child with padding:

```cpp
ImGui::SetCursorPos(GetCursorPos() - style.WindowPadding);
ImGui::BeginChild(...., ImGuiChildFlags_AlwaysUseWindowPadding)
```
